lib/order.rb

```diff
@@ -0,0 +1,81 @@
+require_relative "customer.rb"
+
+class Order
+  attr_accessor :products, :customer, :fulfillment_status
+  attr_reader :id
+
+  def initialize(id, products, customer, fulfillment_status = :pending)
+    @id = id
+    @products = products
+    @customer = customer
+
+    if [:pending, :paid, :processing, :shipped, :complete].include?(fulfillment_status)
+      @fulfillment_status = fulfillment_status
+    else
+      raise ArgumentError, "A valid fulfillment_status was not put into the Order"
+    end
+  end
+
+  def total
+    if @products.length == 0
+      return 0
+    else
+      total = @products.values.reduce(:+)
+      total *= 1.075 #Tax
+      total = total.round(2)
+      return total
+    end
+  end
+
+  def add_product(product_name, price)
+    if @products.keys.include?(product_name)
+      raise ArgumentError, "The product is already in the order"
+    else
+      @products[product_name] = price
+    end
+  end
+
+  def remove_product(product_name)
+    if @products.keys.include?(product_name)
+      @products.reject! { |k| k == product_name }
+    else
+      raise ArgumentError, "The product does not exist in the order"
+    end
+  end
+
+  def self.product_hash(order)
```

Comment:
Nice helper method.  We hadn't gone over `private` at this point, but this would be an excellent `private` method.